### PR TITLE
Fix Itinerary/TripCard visual alignment and a11y

### PR DESCRIPTION
**Pull request by grosbouddha**
- Set time cell width to 48px (was 64px) + 8px of right padding (was 0)
- Fix a11y for title (it's now a SubHeader outside the list, it was a nested li/text)
- Fix departure/arrival city alignement with dot
- Fix multiple :before/:after elements being created due to non selective CSS rule (div/span)

Itinerary:
image
Small itinerary:
image

TripCard:
image

### CODE REVIEW

File: src/itinerary/Itinerary.tsx [hunk 1]
```diff
@@ -16,8 +18,19 @@ interface ItineraryProps {
   readonly headline?: string
 }
 
+
 const isNonEmptyString = (str: string) => isString(str) && str.trim().length > 0
 
+const computeKeyFromPlace = (place: any) => {
```
Comment (pith) [inline src/itinerary/Itinerary.tsx]: Why not using `Place` interface instead of `any` ?

File: src/itinerary/index.tsx [hunk 2]
```diff
@@ -47,58 +44,64 @@ const StyledItinerary = styled(Itinerary)`
...
     border-radius: 50%;
-    top: calc(4px + ${space.m});
-    left: calc(${componentSizes.timeWidth} - 2px);
+    top: calc(4px + ${space.m} + 4px);
+    left: calc(${gutterPaddingStart} + ${componentSizes.timeWidth} - 2px);
+  }
+  
+  & .kirk-itinerary-location.kirk-itinerary--departure .kirk-itinerary-location-city::after,
+  & .kirk-itinerary-location.kirk-itinerary--arrival .kirk-itinerary-location-city::after {
+    top: calc(4px + ${space.m} + 4px);
```
Comment (pith) [inline src/itinerary/index.tsx]: the 4px corresponds to what ?

Comment (grosbouddha) [reply src/itinerary/index.tsx]: It's just ugly adjustments to align stuff. I could use some magic constants to make it feel it was engineered if you prefer. We will need a much deeper refactoring if we want meaningful constants. For now, I will at least merge the 2 4px into one 8px :)

**Comment by grosbouddha** [comment]
FYI, I added some missing padding in the TripCard to align the avatar name with the itinerary city names. (I updated the screenshot in the desc)

**Pull request merged**